contracts/router/BasicRouter.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IBasicRouter} from "./interfaces/IBasicRouter.sol";
+
+import {IERC20} from "@synapseprotocol/sol-lib/contracts/solc8/erc20/IERC20.sol";
+import {IWETH9} from "@synapseprotocol/sol-lib/contracts/universal/interfaces/IWETH9.sol";
+import {SafeERC20} from "@synapseprotocol/sol-lib/contracts/solc8/erc20/SafeERC20.sol";
+
+import {Ownable} from "@openzeppelin/contracts-4.4.2/access/Ownable.sol";
+
+contract BasicRouter is Ownable, IBasicRouter {
+    using SafeERC20 for IERC20;
+
+    address payable public immutable WGAS;
```

Comment:
Should probably add a short docstring/comment for `WGAS` explaining that it's the wrapped ERC20 version of a chain's native currency (ex. WAVAX, WETH, WMOVR)

Reply:
Fixed in [db36df0](https://github.com/synapsecns/synapse-contracts/pull/79/commits/db36df0c5431a5c4c3a5d2e4b37bbb611498a3c6).

The WGAS address itself is used much more often than the actual WGAS contract, so storing as `IWETH9` is suboptimal.